rust/ql/lib/codeql/rust/elements/internal/generated/CallExpr.qll

```diff
@@ -6,28 +6,65 @@
 
 private import codeql.rust.elements.internal.generated.Synth
 private import codeql.rust.elements.internal.generated.Raw
-import codeql.rust.elements.internal.CallExprBaseImpl::Impl as CallExprBaseImpl
+import codeql.rust.elements.ArgList
+import codeql.rust.elements.Attr
 import codeql.rust.elements.Expr
+import codeql.rust.elements.internal.ExprImpl::Impl as ExprImpl
 
 /**
  * INTERNAL: This module contains the fully generated definition of `CallExpr` and should not
  * be referenced directly.
  */
 module Generated {
   /**
-   * A function call expression. For example:
+   * NOTE: Consider using `Call` instead, as that includes all kinds of calls to
+   * functions, and excludes instantiations of tuple structs and tuple enum variants.
+   *
+   * A call expression. For example:
    * ```rust
    * foo(42);
    * foo::<u32, u64>(42);
    * foo[0](42);
    * foo(1) = 4;
+   * Option::Some(42); // tuple enum variant instantiation
    * ```
    * INTERNAL: Do not reference the `Generated::CallExpr` class directly.
    * Use the subclass `CallExpr`, where the following predicates are available.
    */
-  class CallExpr extends Synth::TCallExpr, CallExprBaseImpl::CallExprBase {
+  class CallExpr extends Synth::TCallExpr, ExprImpl::Expr {
```

Comment:
I feel like `CallExpr` may not be the best possible name for this.  I appreciate that you've pointed out in the doc the possibility of using `Call` instead, and that `CallExpr` is probably Rust terminology from some layer or another - but this is still bound to lead to some amount of confusion I think.

Reply:
`CallExpr` comes from the extractor, and it matches [the Rust terminology](https://doc.rust-lang.org/reference/expressions/call-expr.html).